Index: src/router/libpcap/ftmacros.h
===================================================================
--- src/router/libpcap/ftmacros.h
+++ src/router/libpcap/ftmacros.h
@@ -101,6 +101,8 @@
    * this might, for example, be GNU/HURD or one of Debian's kFreeBSD
    * OSes ("GNU/FreeBSD").
    */
+  #ifndef _GNU_SOURCE
   #define _GNU_SOURCE
+  #endif
 
   /*
Index: src/router/libpcap/pcap-linux.c
===================================================================
--- src/router/libpcap/pcap-linux.c
+++ src/router/libpcap/pcap-linux.c
@@ -68,6 +68,8 @@
  */
 
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
 
 #include <config.h>
