Index: src/router/libutils/Makefile
===================================================================
--- src/router/libutils/Makefile	(revision HEAD)
+++ src/router/libutils/Makefile	(working copy)
@@ -403,7 +403,7 @@
 ifeq ($(CONFIG_MADWIFI),y)
-SHOBJS += $(TOP)/register/register_check.o
-WOBJS += $(TOP)/register/register_wifi.o
-OBJS += $(TOP)/register/register_regulatory.o
+# SHOBJS += $(TOP)/register/register_check.o
+# WOBJS += $(TOP)/register/register_wifi.o
+# OBJS += $(TOP)/register/register_regulatory.o
 endif
 ifeq ($(CONFIG_IPV6),y)
 #OBJS += libutils/getifaddrs.o
Index: src/router/libutils/libshutils/shutils.c
===================================================================
--- src/router/libutils/libshutils/shutils.c	(revision HEAD)
+++ src/router/libutils/libshutils/shutils.c	(working copy)
@@ -2215,5 +2215,20 @@
 char *getgptpartitionbyname(const char *dev, const char *name)
 {
 	static char retname[128];
 	return getgptpartitionbyname_safe(retname, dev, name);;
 }
+
+char *hostapd_eap_get_types(void)
+{
+	return "";
+}
+
+void adjust_regulatory(int count)
+{
+	(void)count;
+}
+
+int registered_has_subquarter(void)
+{
+	return 0;
+}
