Index: src/router/pppd/pppd/sys-linux.c
===================================================================
--- src/router/pppd/pppd/sys-linux.c (revision HEAD)
+++ src/router/pppd/pppd/sys-linux.c (working copy)
@@ -122,2 +122,1 @@
-#include <linux/ppp_defs.h>
-#include <linux/if_ppp.h>
+#include <linux/ppp-ioctl.h>
@@ -1407,4 +1406,5 @@
     int u;
     struct pppd_stats *stats;
 {
-    struct ifpppstatsreq req;
+    struct ifreq req;
+    struct ppp_stats data;
@@ -1414,3 +1414,3 @@
-    req.stats_ptr = (caddr_t) &req.stats;
-    strlcpy(req.ifr__name, ifname, sizeof(req.ifr__name));
+    req.ifr_data = (caddr_t) &data;
+    strlcpy(req.ifr_name, ifname, sizeof(req.ifr_name));
     if (ioctl(sock_fd, SIOCGPPPSTATS, &req) < 0) {
@@ -1420,5 +1420,5 @@
-    stats->bytes_in = req.stats.p.ppp_ibytes;
-    stats->bytes_out = req.stats.p.ppp_obytes;
-    stats->pkts_in = req.stats.p.ppp_ipackets;
-    stats->pkts_out = req.stats.p.ppp_opackets;
+    stats->bytes_in = data.p.ppp_ibytes;
+    stats->bytes_out = data.p.ppp_obytes;
+    stats->pkts_in = data.p.ppp_ipackets;
+    stats->pkts_out = data.p.ppp_opackets;
     return 1;
