Index: src/router/rules/busybox.mk
===================================================================
--- src/router/rules/busybox.mk
+++ src/router/rules/busybox.mk
@@ -783,6 +783,7 @@ endif
 	echo "# CONFIG_VMSTAT is not set" >> busybox/.config
 
 	
+	sed -i 's|$$(PREFIX)|$(INSTALLDIR)/busybox|g' busybox/.config
 	cd busybox && make oldconfig KCONFIG_NOTIMESTAMP=1
 
 	if cmp -s busybox/.config busybox/.config.temp ; then \
Index: src/router/rules/configs.mk
===================================================================
--- src/router/rules/configs.mk
+++ src/router/rules/configs.mk
@@ -454,7 +454,7 @@ obj-$(CONFIG_TELNET) += telnet
 obj-$(CONFIG_WDSWATCHDOG) += wdswatchdog
 obj-$(CONFIG_IPV6) += ipv6 dhcpv6
 obj-$(CONFIG_CONNTRACK) += conntrack
-obj-y += radiooff
+obj-$(CONFIG_RADIOOFF) += radiooff
 obj-$(CONFIG_PHP) += libffi zlib libzip libgd libpng libxml2 libmcrypt curl glib20 sqlite php8 zlib
 obj-$(CONFIG_NCURSES) += ncurses
 obj-$(CONFIG_IFTOP) += libpcap iftop
@@ -504,9 +504,11 @@ obj-$(CONFIG_RT3062) += rt3062
 obj-$(CONFIG_RT2860) += rt2860
 obj-$(CONFIG_P910ND) += p910nd
 obj-$(CONFIG_DRIVER_WIRED) += libnfnetlink libnetfilter_log
+ifeq ($(CONFIG_HOSTAPD2),y)
 ifneq ($(CONFIG_OPENSSL),y)
 obj-$(CONFIG_WPA3) += wolfssl
 endif
+endif
 obj-$(CONFIG_WOLFSSL) += wolfssl
 obj-$(CONFIG_HOSTAPD2) += hostapd2
 ifeq ($(KERNELVERSION),6.1)
Index: src/router/rules/hostapd-wps.mk
===================================================================
--- src/router/rules/hostapd-wps.mk
+++ src/router/rules/hostapd-wps.mk
@@ -180,10 +180,8 @@ hostapd2: $(TINY) nvram ubus
 	cp -uvf shared/nl80211.h hostapd-$(HOSTAPDVERSION)/src/drivers/nl80211_copy.h
 	$(MAKE) -C hostapd-$(HOSTAPDVERSION)/hostapd clean
 	$(MAKE) -C hostapd-$(HOSTAPDVERSION)/wpa_supplicant clean
-	echo ` \
-		$(MAKE) -s -C hostapd-$(HOSTAPDVERSION)/hostapd MULTICALL=1 dump_cflags; \
-		$(MAKE) -s -C hostapd-$(HOSTAPDVERSION)/wpa_supplicant MULTICALL=1 dump_cflags \
-	` > hostapd-$(HOSTAPDVERSION)/.cflags.new
+	$(MAKE) -s --no-print-directory -C hostapd-$(HOSTAPDVERSION)/hostapd MULTICALL=1 dump_cflags > hostapd-$(HOSTAPDVERSION)/.cflags.new
+	$(MAKE) -s --no-print-directory -C hostapd-$(HOSTAPDVERSION)/wpa_supplicant MULTICALL=1 dump_cflags >> hostapd-$(HOSTAPDVERSION)/.cflags.new
 	if cmp -s hostapd-$(HOSTAPDVERSION)/.cflags.new hostapd-$(HOSTAPDVERSION)/.cflags ; then \
 		echo "config unchanged" ; \
 	else \
Index: src/router/rules/libpcap.mk
===================================================================
--- src/router/rules/libpcap.mk
+++ src/router/rules/libpcap.mk
@@ -14,8 +14,9 @@ libpcap-configure:
 		--disable-usb \
 		--without-libnl \
 		--disable-canusb --disable-can --disable-bluetooth \
-		CC="$(CC)" CFLAGS="$(COPTS) $(MIPS16_OPT) -std=c99 -D_GNU_SOURCE -fPIC"
+		CC="$(CC)" CFLAGS="$(COPTS) $(MIPS16_OPT) -std=c99 -D_GNU_SOURCE -fPIC -Wno-error=implicit-function-declaration -Wno-implicit-function-declaration"
 libpcap:
+	rm -f libpcap/scanner.c libpcap/scanner.h
 	rm -f $(TOP)/libpcap/libpcap.a
 	rm -f $(TOP)/libpcap/libpcap.so
 	$(MAKE) -C libpcap CC="$(CC)" AR=$(AR) RANLIB=$(RANLIB) libpcap.so
Index: src/router/rules/nettle.mk
===================================================================
--- src/router/rules/nettle.mk
+++ src/router/rules/nettle.mk
@@ -31,8 +31,9 @@ nettle-configure: pcre zlib gmp
 	RANLIB="$(ARCH)-linux-ranlib $(LTOPLUGIN)"
 
 nettle: openssl gmp pcre
+	make -C nettle clean
 	make -C nettle desdata CC_FOR_BUILD="gcc"
-	make -C nettle CC_FOR_BUILD="gcc"
+	make -C nettle CC_FOR_BUILD="gcc" -j4
 
 nettle-clean:
 	-make -C nettle clean
Index: src/router/rules/openssl.mk
===================================================================
--- src/router/rules/openssl.mk
+++ src/router/rules/openssl.mk
@@ -122,7 +122,7 @@ endif
 endif
 
 
-openssl-configure:
+openssl-configure: libucontext
 	cd openssl && CROSS_COMPILE= && ./Configure $(OPENSSL_TARGET) \
 			--prefix=/usr \
 			--libdir=/usr/lib \
@@ -144,7 +144,7 @@ endif
 #endif
 
 	$(MAKE) -C openssl clean
-	-$(MAKE) -C openssl CC="$(CC) -I$(SSLPATH)/crypto -fPIC" MAKEDEPPROG=$(ARCH)-linux-uclibc-gcc $(OPENSSL_MAKEFLAGS)
-	-$(MAKE) -C openssl build_libs CC="$(CC) -fPIC" MAKEDEPPROG=$(ARCH)-linux-uclibc-gcc $(OPENSSL_MAKEFLAGS)
-	-$(MAKE) -C openssl build_programs CC="$(CC) -fPIC" MAKEDEPPROG=$(ARCH)-linux-uclibc-gcc $(OPENSSL_MAKEFLAGS)
+	-$(MAKE) -C openssl CC="$(CC) -I$(SSLPATH)/crypto -fPIC" MAKEDEPPROG=$(ARCH)-linux-uclibc-gcc $(OPENSSL_MAKEFLAGS) -j4
+	-$(MAKE) -C openssl build_libs CC="$(CC) -fPIC" MAKEDEPPROG=$(ARCH)-linux-uclibc-gcc $(OPENSSL_MAKEFLAGS) -j4
+	-$(MAKE) -C openssl build_programs CC="$(CC) -fPIC" MAKEDEPPROG=$(ARCH)-linux-uclibc-gcc $(OPENSSL_MAKEFLAGS) -j4
 	-rm -f openssl/apps/openssl
Index: src/router/rules/pcre.mk
===================================================================
--- src/router/rules/pcre.mk
+++ src/router/rules/pcre.mk
@@ -1,7 +1,7 @@
 pcre-configure:
 	cd pcre && autoreconf -fi
 	cd pcre && ./configure --host=$(ARCH)-linux-uclibc CFLAGS="$(COPTS)  $(MIPS16_OPT)" --prefix=/usr --disable-xmldoc --enable-utf8 --enable-unicode-properties --disable-pcretest-libreadline --libdir=$(TOP)/pcre/.libs
-	touch $(TOP)/pcre/*   
+	sed -i 's/link_all_deplibs=no/link_all_deplibs=yes/g' $(TOP)/pcre/libtool
 
 
 pcre:
Index: src/router/rules/smbd.mk
===================================================================
--- src/router/rules/smbd.mk
+++ src/router/rules/smbd.mk
@@ -9,7 +9,7 @@ smbd-configure: libnl
 	    RANLIB="$(ARCH)-linux-ranlib $(LTOPLUGIN)"
 
 	cd smbd/tools-glib && ./autogen.sh
-	cd smbd/tools-glib && ./configure --prefix=/usr --host=$(ARCH)-linux --disable-shared --enable-static  --libdir=/usr/lib \
+	cd smbd/tools-glib && PKG_CONFIG_PATH=$(TOP)/_staging_static/usr/lib/pkgconfig ./configure --prefix=/usr --host=$(ARCH)-linux --disable-shared --enable-static  --libdir=/usr/lib \
 	    CFLAGS="$(COPTS) $(LTO) $(MIPS16_OPT) $(THUMB) -D_GNU_SOURCE -DNEED_PRINTF -I$(TOP)/_staging_static/usr/include/glib-2.0 -I$(TOP)/_staging_static/usr/lib/glib-2.0/include -Drpl_malloc=malloc -ffunction-sections -fdata-sections -Wl,--gc-sections" \
 	    LDFLAGS="-L$(TOP)/_staging_static/usr/lib $(LDLTO) -ffunction-sections -fdata-sections -Wl,--gc-sections" \
 	    LIBNL_CFLAGS="-I$(TOP)/libnl/include" \
Index: src/router/rules/util-linux.mk
===================================================================
--- src/router/rules/util-linux.mk
+++ src/router/rules/util-linux.mk
@@ -8,6 +8,7 @@ util-linux-configure: ncurses
 	cd util-linux && autoheader
 	cd util-linux && autoreconf -vfi
 	cd util-linux && ./configure --host=$(ARCH)-linux-uclibc --prefix=/usr --libdir=/usr/tmp --disable-gtk-doc CFLAGS="$(COPTS) $(MIPS16_OPT) $(THUMB) -std=c99 -Wno-incompatible-pointer-types -fPIC -DNEED_PRINTF -I$(TOP)/ncurses/include  -ffunction-sections -fdata-sections -Wl,--gc-sections" PKG_CONFIG="/tmp" \
+	--disable-makeinstall-chown \
 	--disable-rpath \
 	--enable-new-mount	\
 	--disable-tls		\
@@ -18,13 +19,14 @@ util-linux-configure: ncurses
 	--without-ncursesw	\
 	--without-tinfo \
 	--without-ncurses
-	make -C util-linux
+	sed -i 's/link_all_deplibs=no/link_all_deplibs=yes/g' util-linux/libtool
+	make -C util-linux -j4
 
 util-linux-clean:
 	make -C util-linux clean
 
 util-linux: util-linux-configure
-	make -C util-linux
+	make -C util-linux -j4
 
 util-linux-install:
 	make -C util-linux install DESTDIR=$(INSTALLDIR)/util-linux
