Index: src/router/shared/utils.h
===================================================================
--- src/router/shared/utils.h
+++ src/router/shared/utils.h
@@ -1455,4 +1455,8 @@
 static inline int has_qam256(const char *prefix)
 {
+#ifdef HAVE_RT2880
+	return 1;
+#else
 	return 0;
+#endif
 }
Index: src/router/shared/wlutils.h
===================================================================
--- src/router/shared/wlutils.h
+++ src/router/shared/wlutils.h
@@ -644,8 +644,12 @@
 extern void free_wifi_clients(struct wifi_client_info *wci);
 extern void free_mac80211_ac(struct mac80211_ac *ac);
 #else
 	#define has_airtime_fairness(prefix) 0
+	#if defined(HAVE_RT2880) && !defined(HAVE_MT76)
+	#define has_shortgi(prefix) 1
+	#else
 	#define has_shortgi(prefix) 0
+	#endif
 	#define has_subeamforming(prefix) 0
 	#define has_mubeamforming(prefix) 0
 	#define has_vht80(interface) 0
