Index: src/router/upnp/src/linux/Makefile
===================================================================
--- src/router/upnp/src/linux/Makefile (revision HEAD)
+++ src/router/upnp/src/linux/Makefile (working copy)
@@ -75,3 +75,3 @@
 TARGET = upnp
-#LIBS = -L$(UPNPBASE) -L$(TOP)/netconf -L$(TOP)/libutils -L$(TOP)/iptables/libiptc \
-#	-L$(TOP)/nvram -L$(TOP)/shared -lupnp -lnetconf -lutils -lshutils -lnvram -liptc $(ADDLIBS)
+#LIBS = -L$(UPNPBASE) -L$(TOP)/libutils -L$(TOP)/iptables/libiptc \
+#	-L$(TOP)/nvram -L$(TOP)/shared -lupnp -lutils -lshutils -lnvram -liptc $(ADDLIBS)
@@ -79,7 +79,7 @@
 ifeq ($(CONFIG_NFTABLES),y)
-LIBS = -L$(UPNPBASE) -L$(TOP)/netconf -L$(TOP)/libutils -L$(TOP)/iptables-new/nftables/libiptc/.libs \
-	-L$(TOP)/nvram -L$(TOP)/shared -lupnp -lnetconf -lutils -lshutils -lnvram -lip4tc $(ADDLIBS)
+LIBS = -L$(UPNPBASE) -L$(TOP)/libutils -L$(TOP)/iptables-new/nftables/libiptc/.libs \
+	-L$(TOP)/nvram -L$(TOP)/shared -lupnp -lutils -lshutils -lnvram -lip4tc $(ADDLIBS)
 else
-LIBS = -L$(UPNPBASE) -L$(TOP)/netconf -L$(TOP)/libutils -L$(TOP)/iptables-new/normal/libiptc/.libs \
-	-L$(TOP)/nvram -L$(TOP)/shared -lupnp -lnetconf -lutils -lshutils -lnvram -lip4tc $(ADDLIBS)
+LIBS = -L$(UPNPBASE) -L$(TOP)/libutils -L$(TOP)/iptables-new/normal/libiptc/.libs \
+	-L$(TOP)/nvram -L$(TOP)/shared -lupnp -lutils -lshutils -lnvram -lip4tc $(ADDLIBS)
 endif
