Index: src/router/rc/rc.c
===================================================================
--- src/router/rc/rc.c	(revision HEAD)
+++ src/router/rc/rc.c	(working copy)
@@ -141,5 +141,6 @@
 #ifdef HAVE_QTN
 	#include "qtn_monitor.c"
 #endif
+#include "switch_mtk_esw.c"
 #include "event.c"
 #include "gratarp.c"
@@ -436,5 +437,6 @@ struct MAIN main_calls[] = {
 #ifdef HAVE_QTN
 	{ "qtn_monitor", NULL, &qtn_monitor_main },
 #endif
+	{ "mtk_esw", NULL, &mtk_esw_main },
 	{ "write", NULL, &write_main },
 	{ "emergency_reboot", NULL, &emergency_reboot_main },
Index: src/router/rc/Makefile
===================================================================
--- src/router/rc/Makefile	(revision HEAD)
+++ src/router/rc/Makefile	(working copy)
@@ -1158,5 +1158,8 @@ endif
 endif
 
+CFLAGS += -DUSE_MTK_GSW
+CFLAGS += -I$(LINUXDIR)/drivers/net/ethernet/raeth
+
 include $(TOP)/rules/generic.mk
 
 CFLAGS += $(DD_CFLAGS)
@@ -1311,6 +1314,7 @@ install: all
 	cd $(INSTALLDIR)/sbin && ln -sf rc gpio
 	cd $(INSTALLDIR)/sbin && ln -sf rc softwarerevision
 	cd $(INSTALLDIR)/sbin && ln -sf rc mtd
+	cd $(INSTALLDIR)/sbin && ln -sf rc mtk_esw
 ifeq ($(CONFIG_GPIOWATCHER),y)
 	cd $(INSTALLDIR)/sbin && ln -sf rc gpiowatcher
 endif

