Index: src/router/services/Makefile
===================================================================
--- src/router/services/Makefile	(revision HEAD)
+++ src/router/services/Makefile	(working copy)
@@ -1641,6 +1641,10 @@ endif
 ifeq ($(CONFIG_DIR860),y)
 CFLAGS_EXTRA += -DHAVE_DIR860
 endif
+ifeq ($(CONFIG_K2P),y)
+CFLAGS_EXTRA += -DHAVE_K2P
+CFLAGS_EXTRA += -I$(TOP)/shared -I$(TOP)/nvram
+endif
 ifeq ($(ARCHITECTURE),broadcom)
 CFLAGS_EXTRA += -DHAVE_BRCMROUTER
 endif
@@ -2182,14 +2185,14 @@ vpath %.c sysinit sysinit/switchlib sysinit/bench services tools networking netw
 all: services_main
 ifeq ($(CONFIG_RT2880),y)
 ifeq ($(CONFIG_MT7620),y)
-	$(CC) $(CFLAGS) -DNEED_PRINTF -o switch tools/switch_gsw.c -DCONFIG_RALINK_MT7620 -L$(TOP)/libutils -lshutils -L$(TOP)/nvram -lnvram
+	$(CC) $(CFLAGS) -DCONFIG_RALINK_MT7621 -I$(LINUXDIR)/drivers/net/ethernet/raeth -DNEED_PRINTF -o switch tools/switch_gsw.c -DCONFIG_RALINK_MT7620 -L$(TOP)/libutils -lshutils -L$(TOP)/nvram -lnvram
 ifeq ($(CONFIG_E1700),y)
 	$(CC) $(CFLAGS) -DNEED_PRINTF -o switch tools/switch_7530.c -DCONFIG_RALINK_MT7620 -L$(TOP)/libutils -lshutils -L$(TOP)/nvram -lnvram
 endif
 else
 	$(CC) $(CFLAGS) -DNEED_PRINTF -o switch tools/switch.c -L$(TOP)/libutils -lshutils -L$(TOP)/nvram -lnvram
 endif
-	$(CC) $(CFLAGS) -DNEED_PRINTF -o mii_mgr tools/mii_mgr.c 
+	$(CC) $(CFLAGS) -DCONFIG_RALINK_MT7621 -I$(LINUXDIR)/drivers/net/ethernet/raeth -DNEED_PRINTF -o mii_mgr tools/mii_mgr.c
 endif
 #	$(CC) $(CFLAGS) $(LDFLAGS) -DNEED_PRINTF -DTEST -o tools/site_survey_test tools/site_survey_broadcom.c -lnvram
 	@echo "[MAKEDEFAULTS] bin/defaults"
