Index: src/router/mac80211/drivers/net/wireless/mediatek/mt76/mt7615/mac.c
===================================================================
--- src/router/mac80211/drivers/net/wireless/mediatek/mt76/mt7615/mac.c	(revision 53571)
+++ src/router/mac80211/drivers/net/wireless/mediatek/mt76/mt7615/mac.c	(working copy)
@@ -920,6 +920,9 @@ void mt7615_mac_sta_poll(struct mt7615_dev *dev)
		msta = list_first_entry(&sta_poll_list, struct mt7615_sta,
					poll_list);
+
+		spin_lock_bh(&dev->sta_poll_lock);
		list_del_init(&msta->poll_list);
+		spin_unlock_bh(&dev->sta_poll_lock);
 
		addr = mt7615_mac_wtbl_addr(dev, msta->wcid.idx) + 19 * 4;
